src/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -260,14 +261,32 @@ private static bool TryResolveConflicts<TItem>(this GraphNode<TItem> root, List<
                     if (acceptedLibraries.TryGetValue(childNode.Key.Name, out acceptedNode) &&
                         childNode != acceptedNode &&
                         childNode.Key.VersionRange != null &&
-                        string.Equals(childNode.Key.TypeConstraint, acceptedNode.Key.TypeConstraint, StringComparison.Ordinal) &&
-                        !childNode.Key.VersionRange.Satisfies(acceptedNode.Item.Key.Version))
+                        string.Equals(
+                            childNode.Key.TypeConstraint, 
+                            acceptedNode.Key.TypeConstraint, 
+                            StringComparison.Ordinal))
                     {
-                        versionConflicts.Add(new VersionConflictResult<TItem>
+                        var versionRange = childNode.Key.VersionRange;
+                        var checkVersion = acceptedNode.Item.Key.Version;
+
+                        // Allow prerelease versions if the selected library is prerelease and the range is
+                        // using the default behavior of filtering to stable versions.
+                        // Ex: [4.0.0, ) should allow 4.0.10-beta if that library was selected during the graph walk
+                        // The decision on if a prerelease version should be allowed should happen previous to this
+                        // check during the walk.
+                        if (checkVersion.IsPrerelease && !versionRange.IncludePrerelease)
```

Comment:
Since VersionRange.SetIncludePrerelease already checks for versionRange.IncludePrerelease, we may not have to check it here